src/data_structures/src/bit_array.cairo

```diff
@@ -1,20 +1,20 @@
-use array::{serialize_array_helper, deserialize_array_helper};
-use byte_array::BYTES_IN_BYTES31_MINUS_ONE;
-use bytes_31::{
-    one_shift_left_bytes_u128, one_shift_left_bytes_felt252, bytes31, BYTES_IN_U128,
-    BYTES_IN_BYTES31,
-};
-use integer::u512;
-use serde::into_felt252_based::SerdeImpl;
+use core::bytes_31::{bytes31};
+use core::integer::u512;
+use core::serde::Serde;
+use core::serde::into_felt252_based::SerdeImpl;
 
 const SELECT_BIT: u128 = 0b10;
+const POW_2_128: felt252 = 0x100000000000000000000000000000000;
+const BYTES_IN_U128: usize = 16;
+const BYTES_IN_BYTES31: usize = 31;
+const BYTES_IN_BYTES31_MINUS_ONE: usize = BYTES_IN_BYTES31 - 1;
 
 #[derive(Clone, Drop)]
-struct BitArray {
-    data: Array<bytes31>,
-    current: felt252,
-    read_pos: usize,
-    write_pos: usize,
+pub struct BitArray {
+    pub data: Array<bytes31>,
+    pub current: felt252,
+    pub read_pos: usize,
+    pub write_pos: usize,
```

Comment:
getters plz, let's try to add a derive in the compiler

Reply:
what's the point of adding a derive if it is just to do the same as accessing it?
Would make sense to declare the field to be read-only I guess ^^'

Reply:
having a getter is completely different from making it pub, if it's pub i can't modify it,  with a getter i can just get the value

Reply:
yeah, I'd imagine more of a new keyword like readonly than adding a new derive, that's what I meant

Reply:
Yeah no derive is definitely the right way. This keyword would make no sense as you need to set the value of your field at some point so it’s not read only

Reply:
then it breaks the `mut` keyword

Reply:
Updated